File: app/src/main/java/com/example/gymlog/data/plan/WorkoutExercise.java
package com.example.gymlog.data.plan;

import com.example.gymlog.data.exercise.Equipment;
import com.example.gymlog.data.exercise.Exercise;
import com.example.gymlog.data.exercise.Motion;
import com.example.gymlog.data.exercise.MuscleGroup;
import com.example.gymlog.data.set.WorkoutSet;

import java.util.List;

public class WorkoutExercise {
    private Exercise exercise; // Базова вправа
    private int exercisesGroupId;
    private List<WorkoutSet> sets;

    public WorkoutExercise(Exercise exercise, int exercisesGroupId, List<WorkoutSet> sets) {
        this.exercise = exercise;
        this.exercisesGroupId = exercisesGroupId;
        this.sets = sets;
    }

    public Exercise getExercise() {
        return exercise;
    }

    public void setExercise(Exercise exercise) {
        this.exercise = exercise;
    }

    // Інші геттери та сеттери

}

File: app/src/main/java/com/example/gymlog/database/room/WorkoutExercise.kt
package com.example.gymlog.database.room

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey



// WorkoutExercise.kt
@Entity(tableName = "WorkoutExercises",
    foreignKeys = [
        ForeignKey(
            entity = WorkoutGymDay::class,
            parentColumns = ["id"],
            childColumns = ["workout_gymday_ID"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Exercise::class,
            parentColumns = ["id"],
            childColumns = ["exerciseId"],
            onDelete = ForeignKey.SET_NULL
        )
    ],
    indices = [                     // ← ДОДАЙ
        Index(value = ["workout_gymday_ID"]),
        Index(value = ["exerciseId"])
    ]
)
data class WorkoutExercise(
    @PrimaryKey(autoGenerate = true)
    val id: Long?,
    val workout_gymday_ID: Long?, // Тип Long для сумісності з батьком
    val exerciseId: Long?,       // nullable, SET_NULL
    val name: String?,
    val description: String?,
    val motion: String?,
    val muscleGroups: String?,
    val equipment: String?
)